modules/lang-painless/src/main/java/org/elasticsearch/painless/action/PainlessExecuteAction.java

```diff
@@ -574,7 +579,9 @@ static Response innerShardOperation(Request request, ScriptService scriptService
                             factory.newFactory(GeoPointFieldScript.CONTEXT.name, request.getScript().getParams(), context.lookup());
                     GeoPointFieldScript geoPointFieldScript = leafFactory.newInstance(leafReaderContext);
                     geoPointFieldScript.runForDoc(0);
-                    return new Response(geoPointFieldScript.asDocValues());
+                    return new Response(Arrays.stream(geoPointFieldScript.asDocValues()).mapToObj(p ->
+                            GeoEncodingUtils.decodeLatitude((int)(p >>> 32)) + ", " +
+                                    GeoEncodingUtils.decodeLongitude((int)(p & 0xFFFFFFFF))).collect(Collectors.toList()));
```

Comment:
@romseygeek can you have a look ? Do you think we could try and reuse some of the value fetcher code here? It feels bad that we try to mimic it but then the two could diverge over time?

Reply:
I think you can use the new `runForDoc(int, Consumer<GeoPoint>)` method on GeoPointFieldScript here, something like:
```
List<GeoPoint> points = new ArrayList<>();
geoPointFieldScript.runForDoc(0, points::add);
return new Response(points);
```

All of these script impls now have a `runForDoc(int, Consumer)` method which can probably be used more generally here.

Reply:
sounds good to me. One more thing: I am no longer sure that re-sorting like doc_values do is important here. Ordering can be tricky, depending on how you use the runtime field: doc_values will resort, fields API does not guarantee ordering, and maybe what we should do here for simulation is return the values in the order that they are emitted. This would make it possible to use the existing runForDoc that takes a Consumer in all the field script classes, and remove the additional asDocValues method. Thoughts?

Reply:
@romseygeek Thank you for the pointer! This will make these things both easier and more consistent. 

@javanna I like this plan a lot assuming we have buy-in from the majority of the team. @jimczi @nik9000 -- how would you both feel about returning the results as they are emitted in the script rather than trying to re-order them? I especially like this for the display results in any kind of Kibana Painless editor because it means the users can immediately tie an emit to a value which makes things easier to debug imo.

Reply:
I like mimicking the `fields` API here. It sorts now because it goes through doc values but it really can do whatever it wants because we say the order is undefined. So I'm fine not sorting, but we should probably also say the order us undefined here. Just because we can spit them out in the order that the script makes them *now* doesn't mean we should require that we do it forever.

Reply:
I think that more than explaining that the order is undefined in painless execute, we could guarantee it here in order of emit, but the main point for users is that in real life the order is different depending on the context: fields, or doc_values etc. so the behaviour is handy and predictable in painless execute but not a guarantee for when the script is used in practice. Maybe you meant the same, but I feel like this is a slightly different perspective